pkg/action/action.go

```diff
@@ -88,5 +104,56 @@ func (c *Conn) Single(ctx context.Context, name string, data json.RawMessage) (j
 	}
 
 	return content.Results[0], nil
+}
+
+// Migrations sends the given migrations command to the backend.
+func (c *Conn) Migrations(ctx context.Context, command string) (json.RawMessage, error) {
+	if c.route != MigrationsRoute {
+		return nil, fmt.Errorf("invalid route for this connection; expected %q, got %q", MigrationsRoute, c.route)
+	}
+
+	addr := c.URL.String()
+	reqBody := struct {
+		Cmd string `json:"cmd"`
+	}{
+		Cmd: command,
+	}
+	encodedBody, err := json.Marshal(reqBody)
+	if err != nil {
+		return nil, fmt.Errorf("marshalling request body: %w", err)
+	}
+	req, err := http.NewRequestWithContext(ctx, "POST", addr, bytes.NewReader(encodedBody))
+	if err != nil {
+		return nil, fmt.Errorf("creating request to backend action service: %w", err)
+	}
+	creds := shared.BasicAuth{
+		Password: c.internalAuthPassword,
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set(shared.AuthHeader, creds.EncPassword())
```

Comment:
I think all of these are basically identical to `Single()`.
You might want to put them into a new method.

Reply:
Yes I know. This kind of refactoring has to be done in an extra step.